.github/workflows/testing.yml

```diff
@@ -73,15 +72,21 @@ jobs:
           python -m pip install --upgrade -r dev-requirements.txt
           python -m pip install --upgrade .
       - name: Try single CLI run of tool
+        env:
+          nvd_api_key: ${{ secrets.NVD_API_KEY }}
         run: |
           NO_EXIT_CVE_NUM=1 python -m cve_bin_tool.cli test/assets/test-kerberos-5-1.15.1.out -n json
       - name: Run async tests
+        env:
+          nvd_api_key: ${{ secrets.NVD_API_KEY }}
         run: >
           pytest -n 4 -v
           --ignore=test/test_cli.py
           --ignore=test/test_cvedb.py
           --ignore=test/test_requirements.py
       - name: Run synchronous tests
+        env:
+          nvd_api_key: ${{ secrets.NVD_API_KEY }}
```

Comment:
Hopefully not needed if we've done our caching and moved tests to longtests, but I wouldn't swear we're done with those so let's keep it.

Reply:
The problem is that long tests are the ones where you're using codecov, so those are the ones I really don't want it to have conditional access to a secret.

There are ways to work around w/ secrets, but they're incredibly risky and on average most of the time everyone (myself included) get them wrong.

It'd be help to have a sense of what kinda of thing the `NVD_API_KEY` is.
* Is it a read-only token used to limit the number of requests to a specific customer?
* Can it make generic writes to a database?
* Is it just for uploading reports? (Can it delete its own reports?)

Reply:
Hm, I'm not sure there's a way around this.  the NVD API Key is needed to get any known vulnerability data from NVD using their API.  (Without it, the requests fail entirely now; previously they were rate limited but didn't fail.)  We have code that gets that data, and tests of that code that we ideally want included in the coverage reports.   

It's not a very valuable secret.  It's used to rate limit our access to the API, so everything using that API key is part of the same pool of requests.  if it leaks then someone else would share our rate limit pool and I guess could denial of service us temporarily and break CI, and it would take a few clicks for me to request a new one (thus expiring the old one), so it doesn't have *no* value, but it probably has very little value to the folk at codecov (who'd generally want their own key), and wouldn't even have much value to someone who compromised codecov.  It can't upload reports, make write statements, or do anything except receive data.

That said, in the spirit of best practices... we could just accept that any NVD requests aren't going to be included in codecov. (I wouldn't want to do it right this week because we've got a summer intern working on coverage improvements and it'll mess with his system but it would be fine in a few weeks once he's done getting data for his final report.)  We've already done a bunch of work so we get a cached version of the database to handle the rest of the tests, so most of the coverage would be ok, and we might be able to mock some other requests to fill the gaps. 

If we go that route, I think we'd need to do some work of sorting out what in long_tests actually needs the key and what doesn't, since they're currently all grouped together.  We're probably going to have to do this eventually because NVD is removing all access methods that don't use the API key in the next year (circa end of 2023) so we'll need to restructure our CI to avoid ever running tests that use the key in PRs, but I wasn't going to worry about that until they make us switch over to API 2.0 which is supposed to come in a couple of months.

Reply:
Ok, so, given _those_ constraints, it is possible to do something.

Offhand, I'd definitely encourage mocking the system. If you have the `push` branch retrieve and cache the nvd db using the secret, then you can use a fake key in the long tests which talks to your mocked service which just feeds the output it has retrieved when it used the real key. That'd enable you to retain coverage testing over the branches while not actually giving a real key to the code. It sounds like this would be the right approach.

It's out of scope for what I'm able to offer (I'm here because my fairly uninteresting PR tripped over this failure and I wanted to offer a basic thing that could get me out of the weeds the next time I visit).

There are fancier things you can do if caching turns out to be a problem (feel free to tag me if you trip on it...).